SpatialGDK/Source/SpatialGDKTests/Public/GDKAutomationTestBase.h

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+#include "Misc/AutomationTest.h"
+#include "SpatialGDKTests/SpatialGDKServices/LocalDeploymentManager/LocalDeploymentManagerUtilities.h"
+
+DEFINE_LOG_CATEGORY_STATIC(LogGDKTestBase, Log, All);
+
+/**
+ * This class extends the Unreal AutomationTestBase to allow unit tests to be augmented with a set-up step, called
+ * before each test.
+ * This class is then offered through a macro, in a similar way to `IMPLEMENT_SIMPLE_AUTOMATION_TEST`.
+ *
+ * To use this test base, the GDK_AUTOMATION_TEST macro should be used, followed by the test body:
+ * ```
+ *	   GDK_AUTOMATION_TEST(MyModule, MyComponent, MyTestName)
+ *	   {
+ *			// do some testing here...
+ *
+ *			return true;
+ *	   }
+ * ```
+ *
+ * Returning `true` indicates a test pass and returning `false` indicates test failure.
+ */
+class FGDKAutomationTestBase : public FAutomationTestBase
+{
+public:
+	FGDKAutomationTestBase(const FString& InName, bool bInComplexTask)
+		: FAutomationTestBase(InName, bInComplexTask)
+	{
+	}
+};
+
+#define GDK_AUTOMATION_TEST(ModuleName, ComponentName, TestName)                                                                           \
+	IMPLEMENT_GDK_AUTOMATION_TEST(TestName, "SpatialGDK." #ModuleName "." #ComponentName "." #TestName)
+
+#define IMPLEMENT_GDK_AUTOMATION_TEST(TestName, PrettyName)                                                                                \
+	class TestName : public FGDKAutomationTestBase                                                                                         \
+	{                                                                                                                                      \
+	public:                                                                                                                                \
+		TestName(const FString& InName)                                                                                                    \
+			: FGDKAutomationTestBase(InName, false)                                                                                        \
+		{                                                                                                                                  \
+		}                                                                                                                                  \
+		virtual uint32 GetTestFlags() const override                                                                                       \
```

Comment:
I feel like we can move MOST of these methods to FGDKAutomationTestBase, so that individual tests can override them if they really want. Also it's nicer to define what we can outside the macro.

Any reason why everything is in the macro that I am missing?

Reply:
No reason, will take a look :)